website/docs/cloud-docs/api-docs/run-tasks/run-task-stages-and-results.mdx

```diff
@@ -0,0 +1,264 @@
+---
+page_title: Run Task Stages and Results - API Docs - Terraform Cloud and Terraform Enterprise
+---
+
+[200]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200
+
+[404]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404
+
+[JSON API document]: /cloud-docs/api-docs#json-api-documents
+
+[JSON API documents]: /cloud-docs/api-docs#json-api-documents
+
+[JSON API error object]: https://jsonapi.org/format/#error-objects
+
+# Run Task Stages API
+
+Each run passes through several [stages](/cloud-docs/run/states) of action, and Terraform Cloud shows the progress through those stages as run states.
+Run task stages are actioned as the run processes the different stages, namely;
+
+- [Post-Plan Stage](/cloud-docs/run/states#3-the-post-plan-stage) which executes [run tasks](/cloud-docs/workspaces/settings/run-tasks) after the plan is complete.
+
+-> **Note:** Run Tasks is a paid feature, available as part of the **Team & Governance** upgrade package. Refer to [Terraform Cloud pricing](https://www.hashicorp.com/products/terraform/pricing) for details.
+
+## Attributes
+
+### Task Stage States
+
+The task stage state is found in `data.attributes.status`, and you can reference the following list of possible states.
+
+| State         | Description                                                  |
+|---------------|--------------------------------------------------------------|
+| `pending`     | The initial status of a task stage once it has been created. |
+| `running`     | The task stage is running.                                   |
+| `passed`      | All of the results in the task stage passed.                 |
+| `failed`      | One more results in the task stage failed.                   |
+| `errored`     | The task stage has errored. This is a final state.           |
+| `canceled`    | The task stage has been canceled.                            |
+| `unreachable` | The task stage could not be executed.                        |
+
+### Task Result States
+
+The task result state is found in `data.attributes.status`, and you can reference the following list of possible states.
+
+| State         | Description                                                   |
+|---------------|---------------------------------------------------------------|
+| `pending`     | The initial status of a task result once it has been created. |
+| `running`     | The task result is running.                                   |
+| `passed`      | All of the results in the task result passed.                 |
+| `failed`      | One more results in the task result failed.                   |
+| `errored`     | The task result has errored. This is a final state.           |
+| `canceled`    | The task result has been canceled.                            |
+| `unreachable` | The task result could not be executed.                        |
```

Comment:
unreachable happens when a stage/result could not be executed in a Run because the Run was stopped prior.

For example. A pre-apply Stage/Result is unreachable if the Run had no changes and was never applied;  That is the run was finished prior to executing the pre-apply phase, so the pre-apply never had the chance to execute.